hr_employee_seniority/models/hr_employee.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# copyright 2013 Michael Telahun Makonnen <mmakonnen@gmail.com>
+# copyright 2017 Denis Leemann, Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import datetime
+
+from odoo import api, fields, models
+from odoo import tools
+
+
+class HrEmployee(models.Model):
+
+    _inherit = 'hr.employee'
+
+    initial_employment_date = fields.Date(
+        string='Initial Date of Employment',
+        help='Date of first employment if it was before the start of the '
+             'first contract in the system.',
+    )
+    length_of_service = fields.Float(
+        'Lenght of Service',
+        compute='_compute_months_service',
+    )
+
+    @api.depends('contract_ids')
+    def _compute_months_service(self):
+        date_now = fields.Date.today()
+        Contract = self.env['hr.contract']
+
+        for employee in self:
+            nb_month = 0
+
+            if employee.initial_employment_date and employee.contract_id:
+                contract = Contract.search(
+                    [('employee_id', '=', employee.id)],
+                    order='date_start asc', limit=1)
+
+                from_dt = fields.Date.from_string(
+                    employee.initial_employment_date)
+                to_dt = fields.Date.from_string(
+                    contract.date_start)
+                nb_month += (to_dt.year - from_dt.year) * 12 + \
```

Comment:
@leemannd I'm not quite sure that this will lead to correct result, tested on runbot and contract from '2016-11-01 to 2016-12-31' will give you only one month of service, from calculation will lead to 12-11=1...which is not correct, since the field is float, can you calculate percentage of days from contract start date to end of the first month, plus percentage of days from end date( or today) untill end month first day...